checks/process.go

```diff
@@ -106,28 +93,106 @@ func (p *ProcessCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.Mess
 	p.lastCPUTime = cpuTimes[0]
 	p.lastRun = time.Now()
 
-	statsd.Client.Gauge("datadog.process.containers.host_count", totalContainers, []string{}, 1)
-	statsd.Client.Gauge("datadog.process.processes.host_count", totalProcs, []string{}, 1)
+	statsd.Client.Gauge("datadog.process.containers.host_count", float64(totalContainers), []string{}, 1)
+	statsd.Client.Gauge("datadog.process.processes.host_count", float64(totalProcs), []string{}, 1)
 	log.Debugf("collected processes in %s", time.Now().Sub(start))
 	return messages, nil
 }
 
+func createProcCtrMessages(
+	procsByCtr map[string][]*model.Process,
+	containers []*model.Container,
+	cfg *config.AgentConfig,
+	sysInfo *model.SystemInfo,
+	groupID int32,
+) ([]model.MessageBody, int, int) {
+	totalProcs, totalContainers := 0, 0
+	msgs := make([]*model.CollectorProc, 0)
+
+	// we first split non-container processes in chunks
+	if procsByCtr[emptyCtrID] != nil {
+		totalProcs += len(procsByCtr[emptyCtrID])
+	}
+	chunks := chunkProcesses(procsByCtr[emptyCtrID], cfg.MaxPerMessage)
+	for _, c := range chunks {
+		msgs = append(msgs, &model.CollectorProc{
+			HostName:  cfg.HostName,
+			Info:      sysInfo,
+			Processes: c,
+			GroupId:   groupID,
+		})
+	}
+
+	ctrProcs := make([]*model.Process, 0)
+	ctrs := make([]*model.Container, 0, len(containers))
+	for _, ctr := range containers {
+		// if all processes are skipped for this container, don't send the container
+		if _, ok := procsByCtr[ctr.Id]; !ok {
+			continue
+		}
+		totalProcs += len(procsByCtr[ctr.Id])
+		totalContainers++
+		ctrProcs = append(ctrProcs, procsByCtr[ctr.Id]...)
+		ctrs = append(ctrs, ctr)
+	}
+
+	if len(ctrs) > 0 {
+		msgs = append(msgs, &model.CollectorProc{
+			HostName:   cfg.HostName,
+			Info:       sysInfo,
+			Processes:  ctrProcs,
+			Containers: ctrs,
+			GroupId:    groupID,
+		})
+	}
+
+	// fill in GroupSize for each CollectorProc and convert them to final messages
+	messages := make([]model.MessageBody, 0, len(msgs))
+	for _, m := range msgs {
```

Comment:
nit: since you're already looping through the list of messages, it might clean things up to move the bookkeeping for counting procs/ctrs here: 

e.g.


```
for _, m := range msgs {
  m.GroupSize = int32(len(msgs))
  messages = append(messages, m)
  totalProcs +=  len(m.Processes)  
  totocalContainers += len(m.Containers)
}
```